kratos/modeler/surrogate_boundary_modeler.h

```diff
@@ -0,0 +1,281 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ariadna Cortes Danes
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/node.h"
+#include "geometries/geometry.h"
+#include "voxel_mesh_generator_modeler.h"
+#include "spatial_containers/geometrical_objects_bins.h"
+#include "utilities/geometry_utilities/nearest_point_utilities.h"
+
+namespace Kratos
+{ 
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class SurrogateBoundaryModeler
+ * @ingroup KratosCore
+ * @brief Generates the voxel mesh plus stores the vector and signed distance from every node to the skin of the model.
+ */
+class SurrogateBoundaryModeler 
```

Comment:
```suggestion
class VoxelMeshSurrogateBoundaryModeler 
```
As this relies on the data structure of the voxel mesher (i.e., doesn't work with a standard unstructured mesh) I'd rename it to make this clear. 

Aside of this, which was the final rationale to make a derived class instead of an optional feature (defaulted to false) in the base voxel mesher? Maybe this is not that bad considering that the extension is not "too long" (we can discuss it in person).

Reply:
Another option could be `VoxelMeshGeneratorWithDistancesModeler`.